File: backend/src/main/java/com/petpro/domain/pet/entity/PetChecklist.java
package com.petpro.domain.pet.entity;

import com.petpro.global.common.entity.BaseTimeEntity;
import jakarta.persistence.*;
import lombok.*;

@Entity
@Table(name = "pet_checklists")
@Getter
@NoArgsConstructor(access = AccessLevel.PROTECTED)
@AllArgsConstructor
@Builder
public class PetChecklist extends BaseTimeEntity {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @OneToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "pet_id", nullable = false, unique = true)
    private Pet pet;

    @Column(name = "friendly_to_strangers", nullable = false)
    private Integer friendlyToStrangers;

    @Column(name = "friendly_to_dogs", nullable = false)
    private Integer friendlyToDogs;

    @Column(name = "friendly_to_cats", nullable = false)
    private Integer friendlyToCats;

    @Column(name = "activity_level", nullable = false)
    private Integer activityLevel;

    @Column(name = "barking_level", nullable = false)
    private Integer barkingLevel;

    @Column(name = "separation_anxiety", nullable = false)
    private Integer separationAnxiety;

    @Column(name = "house_training", nullable = false)
    private Integer houseTraining;

    @Column(name = "command_training", nullable = false)
    private Integer commandTraining;

    @Column(name = "eating_habit", length = 50)
    private String eatingHabit;

    @Column(name = "walk_preference", length = 50)
    private String walkPreference;

    @Column(name = "fear_items", length = 500)
    private String fearItems;

    @Column(name = "additional_notes", columnDefinition = "TEXT")
    private String additionalNotes;

    public void update(Integer friendlyToStrangers, Integer friendlyToDogs, Integer friendlyToCats,
                       Integer activityLevel, Integer barkingLevel, Integer separationAnxiety,
                       Integer houseTraining, Integer commandTraining,
                       String eatingHabit, String walkPreference,
                       String fearItems, String additionalNotes) {
        this.friendlyToStrangers = friendlyToStrangers;
        this.friendlyToDogs = friendlyToDogs;
        this.friendlyToCats = friendlyToCats;
        this.activityLevel = activityLevel;
        this.barkingLevel = barkingLevel;
        this.separationAnxiety = separationAnxiety;
        this.houseTraining = houseTraining;
        this.commandTraining = commandTraining;
        this.eatingHabit = eatingHabit;
        this.walkPreference = walkPreference;
        this.fearItems = fearItems;
        this.additionalNotes = additionalNotes;
    }
}

File: frontend/src/types/petChecklist.ts
/**
 * @fileoverview 반려동물 성향 체크리스트 타입 정의
 * @see docs/develop/pet/frontend.md
 */

/** 성향 체크리스트 */
export interface PetChecklist {
  id: number;
  petId: number;
  friendlyToStrangers: number;
  friendlyToDogs: number;
  friendlyToCats: number;
  activityLevel: number;
  barkingLevel: number;
  separationAnxiety: number;
  houseTraining: number;
  commandTraining: number;
  eatingHabit: string | null;
  walkPreference: string | null;
  fearItems: string | null;
  additionalNotes: string | null;
  createdAt: string;
  updatedAt: string;
}

/** 성향 체크리스트 생성 요청 */
export interface CreatePetChecklistRequest {
  friendlyToStrangers: number;
  friendlyToDogs: number;
  friendlyToCats: number;
  activityLevel: number;
  barkingLevel: number;
  separationAnxiety: number;
  houseTraining: number;
  commandTraining: number;
  eatingHabit?: string;
  walkPreference?: string;
  fearItems?: string;
  additionalNotes?: string;
}

/** 성향 체크리스트 수정 요청 */
export interface UpdatePetChecklistRequest {
  friendlyToStrangers: number;
  friendlyToDogs: number;
  friendlyToCats: number;
  activityLevel: number;
  barkingLevel: number;
  separationAnxiety: number;
  houseTraining: number;
  commandTraining: number;
  eatingHabit?: string;
  walkPreference?: string;
  fearItems?: string;
  additionalNotes?: string;
}

/** 식사 습관 옵션 */
export const EATING_HABIT_OPTIONS = [
  { value: '잘 먹음', label: '잘 먹음' },
  { value: '편식', label: '편식' },
  { value: '소식', label: '소식' },
] as const;

/** 산책 선호도 옵션 */
export const WALK_PREFERENCE_OPTIONS = [
  { value: '좋아함', label: '좋아함' },
  { value: '보통', label: '보통' },
  { value: '싫어함', label: '싫어함' },
] as const;

/** 수치형 항목 라벨 */
export const CHECKLIST_SLIDER_LABELS: Record<string, string> = {
  friendlyToStrangers: '낯선 사람 친화도',
  friendlyToDogs: '다른 강아지 친화도',
  friendlyToCats: '고양이 친화도',
  activityLevel: '활동량',
  barkingLevel: '짖음 정도',
  separationAnxiety: '분리불안',
  houseTraining: '배변 훈련',
  commandTraining: '명령어 훈련',
};
